### ISSUE TITLE
PyCBC does not work with h5py 2.10.0

### ISSUE THREAD

**Issue by duncan-brown** [gwastro/pycbc#2937]
See https://github.com/gwastro/pycbc/pull/2928 for details.

**Comment by spxiwh** [comment]
A further reference from the linked issue is that this doesn't seem a problem on python3.

**Comment by ahnitz** [comment]
I think we need to finally address this. @spxiwh @duncan-brown It does appear to be an issue python3 for me. See the report here https://github.com/gwastro/pycbc/pull/3631/checks?check_run_id=1903398318. 

Our search test *only* runs in python3. I was testing there an issue related to mac installation, and realized it would be easier to unpin h5py. I tried it and it seems to fail with a similar error. 

```
  File &quot;/opt/hostedtoolcache/Python/3.8.7/x64/bin/pycbc_coinc_bank2hdf&quot;, line 104, in &lt;module&gt;
    bankf = bank.TemplateBank(args.bank_file, approximant=args.approximant,
  File &quot;/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pycbc/waveform/bank.py&quot;, line 346, in __init__
    self.ensure_hash()
  File &quot;/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pycbc/waveform/bank.py&quot;, line 373, in ensure_hash
    self.table = self.table.add_fields(template_hash, &apos;template_hash&apos;)
  File &quot;/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pycbc/io/record.py&quot;, line 1262, in add_fields
    newself = add_fields(self, arrays, names=names, assubarray=assubarray)
  File &quot;/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pycbc/io/record.py&quot;, line 465, in add_fields
    return merge_arrays(merge_list, names=names, flatten=True,
  File &quot;/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pycbc/io/record.py&quot;, line 343, in merge_arrays
    new_dt = combine_fields([arr.dtype for arr in merge_list])
  File &quot;/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pycbc/io/record.py&quot;, line 292, in combine_fields
    new_dt = numpy.dtype([dt for dtype in dtypes \
ValueError: invalid shape in fixed-type tuple.
```

**Comment by ahnitz** [comment]
More info here https://github.com/h5py/h5py/issues/1307

**Comment by titodalcanton** [comment]
Fixed by #3632?

**Comment by ahnitz** [comment]
Yes, should be fixed by #3632
